modules/govuk/templates/ckan/nginx.conf.erb

```diff
@@ -1,3 +1,25 @@
+
+<%- if @maintenance_mode -%>
+error_page 503 /scheduled_maintenance.html;
+
+location = /scheduled_maintenance.html {
+  root /usr/share/nginx/www;
+}
+location = /scheduled-maintenance {
+  return 503;
+}
+
+# doing an internal redirect from within a `server` block like this will cause
+# `error_page` to be ignored and an ugly error page to be returned. to get
+# around this, we have to do an *external* redirect to the error page (for
+# any request that *isn't* already requesting the error page, obviously).
+# we *have* to do the redirect from the `server` block because the base template
+# that this config is injected into specifies a number of `location`s itself that
+# we wouldn't otherwise be able to override.
+absolute_redirect off;
+rewrite ^/(?!(scheduled-maintenance))$ /scheduled-maintenance redirect;
```

Comment:
Is this intended to redirect _every_ path, except `/scheduled-maintenance`?

It looks like this would only match `/` at the moment. Perhaps you want:

```suggestion
rewrite ^/(?!(scheduled-maintenance)).*$ /scheduled-maintenance redirect;
```

?

Reply:
That's _exactly_ what it's doing. For the path `/scheduled-maintenance` we want to serve a 503, but for every other path, we want to redirect it to `/scheduled-maintenance`, which gets served by the above `location =`. I know, this is far from ideal but other solutions were worse.

Reply:
Hmmm yes I might have to re-experiment to check this...

Reply:
Yup you're absolutely right.

Reply:
Though entertainingly enough adding the `.*` sends it into an infinite redirect, so I'm going to have to play with this some more...